File: src/main/java/com/lessmarkup/dataobjects/SuccessfulLoginHistory.java
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
package com.lessmarkup.dataobjects;

import com.lessmarkup.interfaces.data.AbstractDataObject;
import com.lessmarkup.interfaces.recordmodel.RequiredField;

import java.time.OffsetDateTime;

public class SuccessfulLoginHistory extends AbstractDataObject {
    private long userId;
    @RequiredField
    private String address;
    @RequiredField
    private OffsetDateTime time;

    /**
     * @return the userId
     */
    public long getUserId() {
        return userId;
    }

    /**
     * @param userId the userId to set
     */
    public void setUserId(long userId) {
        this.userId = userId;
    }

    /**
     * @return the address
     */
    public String getAddress() {
        return address;
    }

    /**
     * @param address the address to set
     */
    public void setAddress(String address) {
        this.address = address;
    }

    /**
     * @return the time
     */
    public OffsetDateTime getTime() {
        return time;
    }

    /**
     * @param time the time to set
     */
    public void setTime(OffsetDateTime time) {
        this.time = time;
    }
}

File: src/main/scala/com/lessmarkup/dataobjects/SuccessfulLoginHistory.scala
/*
 * This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */

package com.lessmarkup.dataobjects

import java.time.OffsetDateTime

import com.lessmarkup.interfaces.annotations.RequiredField
import com.lessmarkup.interfaces.data.DataObject

class SuccessfulLoginHistory extends DataObject {
  var userId: Long = 0
  @RequiredField
  var address: String = null
  @RequiredField
  var time: OffsetDateTime = null
}